packages/web3-eth-ens/src/registry.ts

```diff
@@ -0,0 +1,211 @@
+				})
+				.catch(() => {
+					reject(RevertInstructionError);
+				});
+		});
+		return promise;
+	}
+
+	public async setTTL(
+		name: string,
+		ttl: string,
+		txConfig: NonPayableCallOptions,
+	): Promise<ReceiptInfo | RevertInstructionError> {
+		const promise = new Promise<ReceiptInfo>((resolve, reject) => {
+			this.contract.methods
+				.setTTL(namehash(name), ttl)
+				.send(txConfig)
+				.then(receipt => {
+					resolve(receipt);
+				})
+				.catch(() => {
+					reject(RevertInstructionError);
+				});
+		});
+		return promise;
+	}
+
+	public async setSubnodeOwner(
+		name: string,
+		label: string,
+		address: Address,
+		txConfig: NonPayableCallOptions,
+	): Promise<ReceiptInfo | RevertInstructionError> {
+		const hexStrictLabel = !isHexStrict(label) ? sha3Raw(label) : label;
+
+		const promise = new Promise<TransactionReceipt>((resolve, reject) => {
+			this.contract.methods
+				.setSubnodeOwner(namehash(name), hexStrictLabel, inputAddressFormatter(address))
+				.send(txConfig)
+				.then(receipt => {
+					resolve(receipt);
+				})
+				.catch(() => {
+					reject(RevertInstructionError);
+				});
+		});
+		return promise;
+	}
+
+	public async setSubnodeRecord(
+		name: string,
+		label: string,
+		owner: Address,
+		resolver: Address,
+		ttl: string,
+		txConfig: NonPayableCallOptions,
+	): Promise<ReceiptInfo> {
+		const hexStrictLabel = !isHexStrict(label) ? sha3Raw(label) : label;
+
+		const promise = new Promise<ReceiptInfo>((resolve, reject) => {
+			this.contract.methods
+				.setSubnodeRecord(
+					namehash(name),
+					hexStrictLabel,
+					inputAddressFormatter(owner),
+					inputAddressFormatter(resolver),
+					ttl,
+				)
+				.send(txConfig)
+				.then(receipt => {
+					resolve(receipt);
+				})
+				.catch(() => {
+					reject(RevertInstructionError);
+				});
+		});
+		return promise;
+	}
+
+	public async setApprovalForAll(
+		operator: string,
+		approved: boolean,
+		txConfig: NonPayableCallOptions,
+	): Promise<ReceiptInfo | RevertInstructionError> {
+		return this.contract.methods.setApprovalForAll(operator, approved).send(txConfig);
+	}
+
+	public async isApprovedForAll(owner: Address, operator: Address) {
+		const promise = new Promise((resolve, reject) => {
+			this.contract.methods
+				.isApprovedForAll(inputAddressFormatter(owner), inputAddressFormatter(operator))
+				.call()
+				.then(res => {
+					resolve(res);
+				})
+				.catch(() => {
+					reject(RevertInstructionError);
+				});
+		});
+		return promise;
+	}
+
+	public async recordExists(name: string) {
+		const promise = new Promise((resolve, reject) => {
+			this.contract.methods
+				.recordExists(namehash(name))
+				.call()
+				.then(res => {
+					resolve(res);
+				})
+				.catch(() => {
+					reject(RevertInstructionError);
+				});
+		});
+		return promise;
+	}
+
+	public async getResolver(name: string) {
+		const resolverContract = new Promise((resolve, reject) => {
+			this.contract.methods
+				.getResolver(namehash(name))
```

Comment:
I think according to [registry function](https://github.com/ensdomains/ens/blob/ff0f41747c05f1598973b0fe7ad0d9e09565dfcd/contracts/ENS.sol#L28) it should be 

```suggestion
				.resolver(namehash(name))
```